### ISSUE TITLE
type errors are not checked but tsc does

### ISSUE THREAD

**Issue by scarf005** [microsoft/vscode-typescript-next#114]
## Summary

![image](https://github.com/microsoft/vscode-typescript-next/assets/54838975/ed766e4a-9b99-4a4a-bcc8-a0df23ecad2b)

## Reproduction

[![asciicast](https://asciinema.org/a/645248.svg)](https://asciinema.org/a/645248)

## Description

```ts
const foo: number = null as never

const bar: string = 1

bar = 2
```

1. `foo` and `bar` should cause type error.
2. they are successfully emitting errors on tsc.
3. however, on vscode there aren't any errors issued.

### VSCode
```
Version: 1.87.0
Commit: 019f4d1419fbc8219a181fab7892ebccf7ee29a2
Date: 2024-02-27T23:42:16.599Z
Electron: 27.3.2
ElectronBuildId: 26836302
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Linux x64 6.7.0-060700-generic
```

### Typescript Extension
```
Name: JavaScript and TypeScript Nightly
Id: ms-vscode.vscode-typescript-next
Version: 5.5.20240303
```

**Comment by scarf005** [comment]
it was due to me using [typescript to lua](https://github.com/TypeScriptToLua/vscode-typescript-to-lua) extension.
